packages/components/src/internal/longpress.ts

```diff
@@ -0,0 +1,135 @@
+import { type ElementPart, noChange, nothing } from 'lit';
+import {
+  AsyncDirective,
+  type DirectiveParameters,
+  type Part,
+  PartType,
+  directive,
+} from 'lit/async-directive.js';
+
+type Callbacks = {
+  start: () => void;
+  end: () => void;
+};
+
+/**
+ * A directive that determines,if a HTMLButtonElement is long pressed or only clicked.
+ * Depending on the state it executes the callback once or repeating.
+ *
+ * Usage:
+ *
+ *  <button
+ *    ${longPress({ start: () => this.handleStepUp(), end: () => this.handleChange()})}
+ *  >
+ *    my button
+ *  </button>
+ */
+class LongPressDirective extends AsyncDirective {
+  host: HTMLButtonElement;
+
+  observer: MutationObserver;
+
+  callbacks: Callbacks;
+
+  interval: NodeJS.Timeout;
+
+  timeout: NodeJS.Timeout;
```

Comment:
Not sure on this?

Reply:
What do you mean? I need them to reset timeout and clean up timeouts in the end. Or do you mean the namings?